will/backends/io_adapters/slack.py

```diff
@@ -457,85 +619,72 @@ def _update_channels(self):
             self._channels = channels
             self.save("slack_channel_cache", channels)
 
-    def _update_people(self):
+    def _update_people(self, client=None):
+        "Updates our internal list of Slack users.  Kind of expensive."
         people = {}
-
-        self.handle = self.client.server.username
-
-        for k, v in self.client.server.users.items():
-            user_timezone = None
-            if v.tz:
-                user_timezone = v.tz
-            people[k] = Person(
-                id=v.id,
-                mention_handle="<@%s>" % v.id,
-                handle=v.name,
-                source=clean_for_pickling(v),
-                name=v.real_name,
-            )
-            if v.name == self.handle:
-                self.me = Person(
-                    id=v.id,
-                    mention_handle="<@%s>" % v.id,
-                    handle=v.name,
-                    source=clean_for_pickling(v),
-                    name=v.real_name,
-                )
-            if user_timezone and user_timezone != 'unknown':
-                people[k].timezone = user_timezone
-                if v.name == self.handle:
-                    self.me.timezone = user_timezone
+        if client:
+            for page in client.users_list(limit=self.PAGE_LIMIT):
+                for member in page["members"]:
+                    if member["deleted"]:
+                        continue
+                    member_id = member["id"]
+                    user_timezone = member.get("tz")
+                    people[member_id] = Person(
+                        id=member_id,
+                        mention_handle=member.get("mention_handle", "SECRETAGENTMAN"),
```

Comment:
I love having this as the fallback value but we should probably use an empty string here and at line 637.

Reply:
Removed